src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -110,6 +113,17 @@ public SnowflakeSinkTask() {
     totalTaskCreationCount =
         totalTaskCreationCount != -1 ? totalTaskCreationCount + 1 : totalTaskCreationCount;
     this.taskOpenCount = 0;
+    this.ingestSdkProvider = SnowflakeSinkConnector.ingestSdkProvider;
+  }
+
+  /**
+   * TEST ONLY
+   * the ingestSdkProvider should be initialized before task creation begins. Since we can't control what is passed to this task (extends kafka sink task), the provider must be static and thus un-mockable
+   * **/
+  @VisibleForTesting
+  public SnowflakeSinkTask(IngestSdkProvider ingestSdkProvider) {
```

Comment:
couldn't figure out a better way to pass the provider in for testing, open to suggestions